src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryFunction.java

```diff
@@ -126,8 +125,7 @@ public RepositoryDirectoryValue.Builder fetch(
       Map<RepoRecordedInput, String> recordedInputValues,
       SkyKey key)
       throws RepositoryFunctionException, InterruptedException {
-    if (workerExecutorService == null
-        || env.inErrorBubblingForSkyFunctionsThatCanFullyRecoverFromErrors()) {
+    if (!useWorkers || env.inErrorBubblingForSkyFunctionsThatCanFullyRecoverFromErrors()) {
```

Comment:
Maybe this change also makes callig `inErrorBubblingForSkyFunctionsThatCanFullyRecoverFromErrors()` unnecessary?

Reply:
I'd rather not poke that bear until absolutely necessary, since that one was also hard to reproduce.

Reply:
update: I poked the bear. Fairly sure that this is fixed.